vimeo-networking/src/main/java/com/vimeo/networking/model/Comment.java

```diff
@@ -75,12 +90,126 @@ public boolean equals(Object o) {
 
         Comment that = (Comment) o;
 
-        return ((this.uri != null && that.uri != null) ? this.uri.equals(that.uri) : false);
+        return ((this.uri != null && that.uri != null) && this.uri.equals(that.uri));
     }
 
     @Override
     public int hashCode() {
         return this.uri != null ? this.uri.hashCode() : 0;
     }
 
+    public static class CommentTypeAdapter extends TypeAdapter<Comment> {
+
+        private final static EnumTypeAdapter<CommentType> sCommentTypeEnumTypeAdapter =
+                new EnumTypeAdapter<>(CommentType.class);
+
+        @NotNull
+        private final UserTypeAdapter mUserTypeAdapter;
+        @NotNull
+        private final MetadataTypeAdapter mMetadataTypeAdapter;
+
+        public CommentTypeAdapter(@NotNull UserTypeAdapter userTypeAdapter,
+                                  @NotNull MetadataTypeAdapter metadataTypeAdapter) {
+            mUserTypeAdapter = userTypeAdapter;
+            mMetadataTypeAdapter = metadataTypeAdapter;
+        }
+
+        @Override
+        public void write(JsonWriter out, Comment value) throws IOException {
+            out.beginObject();
+            if (value == null) {
+                ClientLogger.d("Comment null in write()");
+                out.endObject();
+                return;
+            }
+            if (value.uri != null) {
+                out.name("uri").value(value.uri);
+            }
+            if (value.type != null) {
+                out.name("type");
+                sCommentTypeEnumTypeAdapter.write(out, value.type);
+            }
+            if (value.createdOn != null) {
+                out.name("created_on").value(ISO8601.fromDate(value.createdOn));
+            }
+            if (value.text != null) {
+                out.name("text").value(value.text);
+            }
+            if (value.user != null) {
+                out.name("user");
+                mUserTypeAdapter.write(out, value.user);
+            }
+            if (value.metadata != null) {
+                out.name("metadata");
+                mMetadataTypeAdapter.write(out, value.metadata);
+            }
+
+            out.endObject();
+        }
+
+        @Override
+        public Comment read(JsonReader in) throws IOException {
+            final Comment comment = new Comment();
+            in.beginObject();
+            while (in.hasNext()) {
+                String nextName = in.nextName();
+                JsonToken jsonToken = in.peek();
+                if (jsonToken == JsonToken.NULL) {
+                    in.skipValue();
+                    continue;
+                }
```

Comment:
We could maybe have a static method that we call from within the while loop, but I think that is all we can abstract. Every object is unique, and we can't assume each has the same design. Some could be arrays, some objects, etc.